### PR TITLE
feat(datepicker): add positionTarget property

### LINKED ISSUE #3092: Feature request: custom datepicker's popup target for positioning
### Feature request for datepicker

1. Make the target element that is used to determine the datepicker's popup position customisable (e.g. `positionTarget` input property), now [it's a hard-coded input](https://github.com/ng-bootstrap/ng-bootstrap/blob/5444d133d100c36a6f238c1e1f77972c427f3fcf/src/datepicker/datepicker-input.ts#L214);
2. Apply & update the corresponding css class (`.ngb-dp-bottom`, `.ngb-dp-top`, `.ngb-dp-left`, `.ngb-dp-right`) based on the actual datepicker's popup position.

### Motivation

This will let to achieve similar things to this (pay attention to the popup arrow and position):

image

Maybe this is not very necessary for the original bootstrap but It can be useful for the themed versions (latter my case).

If agreed - I can make a PR for this.

### PR DESCRIPTION
**Pull request by dzonatan**
Partially closes #3092.

### CODE REVIEW

File: demo/src/app/components/datepicker/datepicker.module.ts [hunk 1]
```diff
@@ -106,6 +108,12 @@ const DEMOS = {
     type: NgbdDatepickerConfig,
     code: require('!!raw-loader!./demos/config/datepicker-config'),
     markup: require('!!raw-loader!./demos/config/datepicker-config.html')
+  },
+  positiontarget: {
```
Comment (maxokorokov) [inline demo/src/app/components/datepicker/datepicker.module.ts]: Could you please place this **before** the "config" demo for consistency → "config" is always last demo for all components.

File: src/datepicker/datepicker-input.ts [hunk 2]
```diff
@@ -204,6 +204,14 @@ export class NgbInputDatepicker implements OnChanges,
    */
   @Input() container: string;
 
+  /**
+   * A selector or html element specifying the element the datepicker popup should be positioned by.
```
Comment (maxokorokov) [inline src/datepicker/datepicker-input.ts]: selector → css selector ? (if you remove the second redundant line)
by → against ?

File: src/datepicker/datepicker-input.ts [hunk 3]
```diff
@@ -204,6 +204,14 @@ export class NgbInputDatepicker implements OnChanges,
...
+   * Supports css selector or reference of the html element.
```
Comment (maxokorokov) [inline src/datepicker/datepicker-input.ts]: this is kinda redundant

**Comment by dzonatan** [comment]
Comments fixed.

**Comment by maxokorokov** [comment]
Thanks @dzonatan!

**Pull request merged**